cli/cmd/proxy/list/command_test.go

```diff
@@ -0,0 +1,211 @@
+			namespace: "default",
+			pods:      []v1.Pod{},
+		},
+		"Gateway pods": {
+			namespace: "default",
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "ingress-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							"component": "ingress-gateway",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "mesh-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							"component": "mesh-gateway",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "terminating-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							"component": "terminating-gateway",
+						},
+					},
+				},
+			},
+		},
+		"API Gateway Pods": {
+			namespace: "default",
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "api-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							"api-gateway.consul.hashicorp.com/managed": "true",
+						},
+					},
+				},
+			},
+		},
+		"Sidecar Pods": {
+			namespace: "default",
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "sidecar",
+						Namespace: "default",
+						Labels: map[string]string{
+							"consul.hashicorp.com/connect-inject-status": "injected",
+						},
+					},
+				},
+			},
+		},
+		"All kinds of Pods": {
+			namespace: "default",
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "sidecar",
+						Namespace: "default",
+						Labels: map[string]string{
+							"consul.hashicorp.com/connect-inject-status": "injected",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "mesh-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							"component": "mesh-gateway",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "api-gateway",
+						Namespace: "default",
+						Labels: map[string]string{
+							"api-gateway.consul.hashicorp.com/managed": "true",
+						},
+					},
+				},
+			},
+		},
+		"Pods in multiple namespaces": {
+			namespace: "",
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "api-gateway",
+						Namespace: "consul",
+						Labels: map[string]string{
+							"api-gateway.consul.hashicorp.com/managed": "true",
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "sidecar",
+						Namespace: "default",
+						Labels: map[string]string{
+							"consul.hashicorp.com/connect-inject-status": "injected",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			client := fake.NewSimpleClientset()
+
+			// Add the pods to the client.
+			for _, pod := range tc.pods {
+				client.CoreV1().Pods(pod.ObjectMeta.Namespace).Create(context.Background(), &pod, metav1.CreateOptions{})
+			}
+
```

Comment:
`fake.NewSimpleClientset` will take in objects so you don't need to do the corev1 Create()!
```suggestion
			client := fake.NewSimpleClientset(tc.pods...)
```
(double-check my syntax here please!)

Reply:
Oh yeah! I forgot about that.